### PR TITLE
Renamed PublishingRequestResolver to TicketResolver and moved persist…

### PR DESCRIPTION
**Pull request by monaullah**
…Ticket from handler to resolver. Added functionality and tests for error handling for publishing workflow.

### CODE REVIEW

**Review by anetteOlli — APPROVED** [review]
Kanskje litt mer clean-code stil, men ellers bra.

File: tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            var customerTransactionResult = getCustomerTransactionResult(customerId);
+
+            if (customerTransactionResult.customerAllowsRegistratorsToPublishDataAndMetadata()) {
+                ((PublishingRequestCase) ticket).setWorkflow(PublishingWorkflow.REGISTRATOR_PUBLISHES_METADATA_AND_FILES);
+                var persistedTicket = persistTicket(ticket);
+                approveTicketAndPublishPublication(persistedTicket, publication);
+                return persistedTicket;
+            }
+        }
+        return persistTicket(ticket);
```
Comment (truhacevkir) [inline tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java:61]: Bra, jeg hadde kanskje forkortet denne metoden noe, for eksempel:

    public TicketEntry resolveAndPersistTicket(TicketEntry ticket, Publication publication, URI customerId) throws ApiGatewayException {
        if (isPublishingRequestCase(ticket)) {
            return resolvePublishingRequest(ticket, publication, customerId);
        }
        return persistTicket(ticket);
    }

**Review by truhacevkir — APPROVED** [review]

File: tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            var publishingRequestCase = (PublishingRequestCase) ticket;
+            publishingRequestCase.setWorkflow(customerTransactionResult.convertToPublishingWorkflow());
+
+            if (REGISTRATOR_PUBLISHES_METADATA_AND_FILES.equals(publishingRequestCase.getWorkflow())) {
+                var persistedTicket = persistTicket(ticket);
+                approveTicketAndPublishPublication(persistedTicket, publication);
+                return persistedTicket;
+            }
+        }
+        return persistTicket(ticket);
```
Comment (anetteOlli) [inline tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java:63]: Her blir vel ticketen persistert to ganger hvis det er en REGISTRATOR_PUBLISHES_METADATA_AND_FILES?
persistTicket før if-sjekken?

**Review by anetteOlli — CHANGES_REQUESTED** [review]

File: tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java (hunk 2 above)
Comment (monaullah) [reply tickets/src/main/java/no/unit/nva/publication/ticket/create/TicketResolver.java:63]: Den blir returnet i if-en

**Review by anetteOlli — APPROVED** [review]

File: tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java [hunk 3]
```diff
@@ -262,9 +263,9 @@ void shouldUpdateExistingDoiRequestWhenNewDoiIsRequestedButUnfulfilledDoiRequest
         var publication = createPersistedPublishedPublication();
         var owner = UserInstance.fromPublication(publication);
         var requestBody = constructDto(DoiRequest.class);
-        publishingRequestResolver = new PublishingRequestResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithUnresolvableClient(),
+        ticketResolver = new TicketResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithUnresolvableClient(),
```
Comment (StigNorland) [inline tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java:266]: Denne settes i  
```
@BeforeEach
    public void setup(...)

```    
unødvendig å init to ganger i samme kall.

File: tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java [hunk 4]
```diff
@@ -273,9 +274,9 @@ void shouldUpdateExistingDoiRequestWhenNewDoiIsRequestedButUnfulfilledDoiRequest
         var createdTicket = fetchTicket(response).copy();
         var secondRequest = createHttpTicketCreationRequest(requestBody, publication, owner);
         output = new ByteArrayOutputStream();
-        publishingRequestResolver = new PublishingRequestResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithUnresolvableClient(),
+        ticketResolver = new TicketResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithUnresolvableClient(),
```
Comment (StigNorland) [inline tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java:277]: same same

File: tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java [hunk 5]
```diff
@@ -341,25 +342,57 @@ void shouldPublishPublicationAndSetTicketStatusToApprovedWhenCustomerAllowsPubli
         var publication = TicketTestUtils.createPersistedPublication(PublicationStatus.DRAFT, resourceService);
         var requestBody = constructDto(PublishingRequestCase.class);
         var owner = UserInstance.fromPublication(publication);
-        publishingRequestResolver = new PublishingRequestResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithPublisherAllowingPublishing(),
+        ticketResolver = new TicketResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithPublisherAllowingPublishing(),
```
Comment (StigNorland) [inline tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java:345]: same same

File: tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java [hunk 6]
```diff
@@ -341,25 +342,57 @@ void shouldPublishPublicationAndSetTicketStatusToApprovedWhenCustomerAllowsPubli
...
         assertThat(getTicketStatusForPublication(publication), is(equalTo(TicketStatus.COMPLETED)));
     }
 
+    @Test
+    void shouldSetTicketWorkflowWhenCustomerAllowsPublishingMetadataOnly()
+            throws ApiGatewayException, IOException {
+        var publication = TicketTestUtils.createPersistedPublication(PublicationStatus.DRAFT, resourceService);
+        var requestBody = constructDto(PublishingRequestCase.class);
+        var owner = UserInstance.fromPublication(publication);
+        ticketResolver = new TicketResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithCustomerAllowingPublishingMetadataOnly(),
```
Comment (StigNorland) [inline tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java:361]: same same

File: tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java [hunk 7]
```diff
@@ -341,25 +342,57 @@ void shouldPublishPublicationAndSetTicketStatusToApprovedWhenCustomerAllowsPubli
...
+        assertThat(getTicketPublishingWorkflow(publication), is(equalTo(PublishingWorkflow.REGISTRATOR_PUBLISHES_METADATA_ONLY)));
+    }
+
+    @Test
+    void shouldSetTicketWorkflowWhenCustomerRequiresApprovalForMetadataAndFiles()
+            throws ApiGatewayException, IOException {
+        var publication = TicketTestUtils.createPersistedPublication(PublicationStatus.DRAFT, resourceService);
+        var requestBody = constructDto(PublishingRequestCase.class);
+        var owner = UserInstance.fromPublication(publication);
+        ticketResolver = new TicketResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithPublisherRequiringApproval(),
```
Comment (StigNorland) [inline tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java:377]: same same

File: tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java [hunk 8]
```diff
@@ -341,25 +342,57 @@ void shouldPublishPublicationAndSetTicketStatusToApprovedWhenCustomerAllowsPubli
...
+    }
+
     @Test
     void shouldPublishPublicationAndFileAndSetTicketStatusToApprovedWhenCustomerAllowsPublishing()
         throws ApiGatewayException, IOException {
         var publication = TicketTestUtils.createPersistedPublicationWithUnpublishedFiles(PublicationStatus.DRAFT, resourceService);
         var requestBody = constructDto(PublishingRequestCase.class);
         var owner = UserInstance.fromPublication(publication);
-        publishingRequestResolver = new PublishingRequestResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithPublisherAllowingPublishing(),
+        ticketResolver = new TicketResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithPublisherAllowingPublishing(),
```
Comment (StigNorland) [inline tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java:393]: same same

File: tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java [hunk 9]
```diff
@@ -370,18 +403,41 @@ void shouldPublishPublicationAndFileAndSetTicketStatusToApprovedWhenCustomerAllo
...
     @Test
-    void shouldIgnorePublishingRequestWhenItCanNotBeResolved() throws ApiGatewayException, IOException {
+    void shouldReturnBadGatewayWhenHttpClientWithNonResolvablePublishingWorkflow() throws ApiGatewayException, IOException {
         var publication = TicketTestUtils.createPersistedPublication(PublicationStatus.DRAFT, resourceService);
         var requestBody = constructDto(PublishingRequestCase.class);
         var owner = UserInstance.fromPublication(publication);
-        publishingRequestResolver = new PublishingRequestResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithNonResolvedPublishingWorkflow(),
-                                                                                                                  secretsManagerClient));
-        this.handler = new CreateTicketHandler(ticketService, resourceService, publishingRequestResolver);
+        ticketResolver = new TicketResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithNonResolvedPublishingWorkflow(),
```
Comment (StigNorland) [inline tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java:410]: her også

File: tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java [hunk 10]
```diff
@@ -370,18 +403,41 @@ void shouldPublishPublicationAndFileAndSetTicketStatusToApprovedWhenCustomerAllo
...
+        assertThat(problem.getDetail(), is(equalTo("Unable to resolve customer publishing workflow")));
+        assertThat(resourceService.getPublication(publication).getStatus(), is(equalTo(PublicationStatus.DRAFT)));
+    }
+
+    @Test
+    void shouldReturnBadGatewayWhenHttpClientUnableToRetrievePublishingWorkflow() throws ApiGatewayException, IOException {
+        var publication = TicketTestUtils.createPersistedPublication(PublicationStatus.DRAFT, resourceService);
+        var requestBody = constructDto(PublishingRequestCase.class);
+        var owner = UserInstance.fromPublication(publication);
+        ticketResolver = new TicketResolver(resourceService, ticketService, getUriRetriever(getHttpClientWithUnresolvableClient(),
```
Comment (StigNorland) [inline tickets/src/test/java/no/unit/nva/publication/ticket/create/CreateTicketHandlerTest.java:429]: her også

**Pull request merged**